### ISSUE TITLE
rclone v1.61.0 serve webdav --baseurl is broken

### ISSUE THREAD

**Issue by wuhgit** [rclone/rclone#6650]
#### The associated forum post URL from `https://forum.rclone.org`

NONE

#### What is the problem you are having with rclone?

Server: rclone v1.61.0 serve webdav
Client: rclone v1.61.0 mount webdav from server

After upgrade to v1.61.0, the client cannot mount server's webdav, the logs show that the root directory cannot be found.
`webdav root '': Statfs failed: Not Found: 404 Not Found` .
but I can be accessed the webdav through the browser.

But after I rolled back the server rclone to 1.60.1, everything works fine again.

#### What is your rclone version (output from `rclone version`)

rclone v1.61.0

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

both server and client are Arch Linux x86_64

#### Which cloud storage system are you using? (e.g. Google Drive)

Local disk

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

Server: 

```
/usr/bin/rclone serve webdav /mnt/ \
                    --baseurl "/webdav" \
                    --htpasswd /etc/rclone/rclone_webdav_auth \
                    --dir-cache-time 5s \
                    --log-level ERROR
```

Client: 

```
/usr/bin/rclone mount server: /mnt/cloud/server/ \
                                                  --config /etc/rclone/rclone.conf \
                                                  --allow-other \
                                                  --allow-non-empty \
                                                  --vfs-cache-mode writes \
                                                  --cache-dir /tmp/rclone/ \
                                                  --vfs-cache-max-age 10m \
                                                  --vfs-cache-max-size 2G \
                                                  --use-mmap \
                                                  --umask 0000 \
                                                  --transfers 2 \
                                                  --buffer-size 64M \
                                                  --dir-cache-time 30s \
                                                  --timeout 120m
```

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

```
2022/12/22 16:22:43 DEBUG : Setting --config "/etc/rclone/rclone.conf" from environment variable RCLONE_CONFIG="/etc/rclone/rclone.conf"
<7>DEBUG : rclone: Version "v1.61.0" starting with parameters ["/usr/bin/rclone" "-vv" "mount" "server:" "/mnt/cloud/server/" "--config" "/etc/rclone/rclone.conf" "--allow-other" "--allow-non-empty" "--vfs-cache-mode" "writes" "--cache-dir" "/tmp/rclone/" "--vfs-cache-max-age" "10m" "--vfs-cache-max-size" "2G" "--use-mmap" "--umask" "0000" "--transfers" "2" "--buffer-size" "64M" "--dir-cache-time" "30s" "--timeout" "120m"]
<7>DEBUG : rclone: systemd logging support activated
<7>DEBUG : Creating backend with remote "server:"
<7>DEBUG : Using config file from "/etc/rclone/rclone.conf"
<7>DEBUG : found headers: 
<6>INFO  : webdav root '': poll-interval is not supported by this remote
<7>DEBUG : vfs cache: root is "/tmp/rclone"
<7>DEBUG : vfs cache: data root is "/tmp/rclone/vfs/server"
<7>DEBUG : vfs cache: metadata root is "/tmp/rclone/vfsMeta/server"
<7>DEBUG : Creating backend with remote "/tmp/rclone/vfs/server/"
<7>DEBUG : fs cache: renaming cache item "/tmp/rclone/vfs/server/" to be canonical "/tmp/rclone/vfs/server"
<7>DEBUG : Creating backend with remote "/tmp/rclone/vfsMeta/server/"
<7>DEBUG : fs cache: renaming cache item "/tmp/rclone/vfsMeta/server/" to be canonical "/tmp/rclone/vfsMeta/server"
<7>DEBUG : webdav root '': Mounting on "/mnt/cloud/server/"
<6>INFO  : vfs cache: cleaned: objects 0 (was 0) in use 0, to upload 0, uploading 0, total size 0 (was 0)
<7>DEBUG : : Root: 
<7>DEBUG : : >Root: node=/, err=<nil>
<7>DEBUG : /: Attr: 
<7>DEBUG : /: >Attr: attr=valid=1s ino=0 size=0 mode=drwxrwxrwx, err=<nil>
<7>DEBUG : : Statfs: 
<3>ERROR : webdav root '': Statfs failed: Not Found: 404 Not Found
<7>DEBUG : : >Statfs: stat={Blocks:4503599627370495 Bfree:4503599627370495 Bavail:4503599627370495 Files:1000000000 Ffree:1000000000 Bsize:4096 Namelen:255 Frsize:4096}, err=<nil>
<7>DEBUG : /: ReadDirAll: 
<7>DEBUG : /: >ReadDirAll: item=2, err=<nil>
<7>DEBUG : /: Attr: 
<7>DEBUG : /: >Attr: attr=valid=1s ino=0 size=0 mode=drwxrwxrwx, err=<nil>
<7>DEBUG : /: Attr: 
<7>DEBUG : /: >Attr: attr=valid=1s ino=0 size=0 mode=drwxrwxrwx, err=<nil>
<7>DEBUG : : Statfs: 
<7>DEBUG : : >Statfs: stat={Blocks:274877906944 Bfree:274877906944 Bavail:274877906944 Files:1000000000 Ffree:1000000000 Bsize:4096 Namelen:255 Frsize:4096}, err=<nil>
<7>DEBUG : : Statfs: 
<7>DEBUG : : >Statfs: stat={Blocks:274877906944 Bfree:274877906944 Bavail:274877906944 Files:1000000000 Ffree:1000000000 Bsize:4096 Namelen:255 Frsize:4096}, err=<nil>
<7>DEBUG : : Statfs: 
<7>DEBUG : : >Statfs: stat={Blocks:274877906944 Bfree:274877906944 Bavail:274877906944 Files:1000000000 Ffree:1000000000 Bsize:4096 Namelen:255 Frsize:4096}, err=<nil>
<7>DEBUG : /: ReadDirAll: 
<7>DEBUG : /: >ReadDirAll: item=2, err=<nil>
<7>DEBUG : /: Attr: 
<7>DEBUG : /: >Attr: attr=valid=1s ino=0 size=0 mode=drwxrwxrwx, err=<nil>
<7>DEBUG : : Statfs: 
<7>DEBUG : : >Statfs: stat={Blocks:274877906944 Bfree:274877906944 Bavail:274877906944 Files:1000000000 Ffree:1000000000 Bsize:4096 Namelen:255 Frsize:4096}, err=<nil>
<7>DEBUG : /: Attr: 
<7>DEBUG : /: >Attr: attr=valid=1s ino=0 size=0 mode=drwxrwxrwx, err=<nil>
<7>DEBUG : : Statfs: 
<7>DEBUG : : >Statfs: stat={Blocks:274877906944 Bfree:274877906944 Bavail:274877906944 Files:1000000000 Ffree:1000000000 Bsize:4096 Namelen:255 Frsize:4096}, err=<nil>
<7>DEBUG : /: ReadDirAll: 
<7>DEBUG : /: >ReadDirAll: item=2, err=<nil>
<7>DEBUG : /: Attr: 
<7>DEBUG : /: >Attr: attr=valid=1s ino=0 size=0 mode=drwxrwxrwx, err=<nil>
<7>DEBUG : : Statfs: 
<7>DEBUG : : >Statfs: stat={Blocks:274877906944 Bfree:274877906944 Bavail:274877906944 Files:1000000000 Ffree:1000000000 Bsize:4096 Namelen:255 Frsize:4096}, err=<nil>
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by darthShadow** [comment]
Please share a redacted config and the logs from the `serve webdav` command.

**Comment by wuhgit** [comment]
sorry, I forgot the log from the server side.

```
/usr/bin/rclone -vv serve webdav /mnt/ \
                                   --baseurl "/webdav" \
                                   --addr :32123 \
                                   --htpasswd /etc/rclone/rclone_webdav_auth \
                                   --dir-cache-time 5s
```

```
2022/12/22 17:59:06 DEBUG : Setting --config "/etc/rclone/rclone.conf" from environment variable RCLONE_CONFIG="/etc/rclone/rclone.conf"
2022/12/22 17:59:06 DEBUG : rclone: Version "v1.61.0" starting with parameters ["/usr/bin/rclone" "-vv" "serve" "webdav" "/mnt/" "--baseurl" "/webdav" "--htpasswd" "/etc/rclone/rclone_webdav_auth" "--dir-cache-time" "5s"]
2022/12/22 17:59:06 DEBUG : Creating backend with remote "/mnt/"
2022/12/22 17:59:06 DEBUG : Using config file from "/etc/rclone/rclone.conf"
2022/12/22 17:59:06 DEBUG : fs cache: renaming cache item "/mnt/" to be canonical "/mnt"
2022/12/22 17:59:06 INFO  : Local file system at /mnt: poll-interval is not supported by this remote
2022/12/22 17:59:06 INFO  : Using "/etc/rclone/rclone_webdav_auth" as htpasswd storage
2022/12/22 17:59:06 NOTICE: Local file system at /mnt: WebDav Server started on [http://[::]:32123/webdav/]
2022/12/22 17:59:15 INFO  : /: PROPFIND from 127.0.0.1:60458
2022/12/22 17:59:15 INFO  : /: PROPFIND from [::1]:49610
2022/12/22 17:59:49 INFO  : /: PROPFIND from 127.0.0.1:60876
2022/12/22 18:03:06 INFO  : /: PROPFIND from [::1]:34644
2022/12/22 18:03:06 INFO  : /: PROPFIND from 127.0.0.1:58650
2022/12/22 18:04:06 INFO  : /: PROPFIND from [::1]:57396
```

this is a complete log from the server and nothing anymore.

**Comment by wuhgit** [comment]
server side does not have any config.

client side config:

```
[server]
type = webdav
url = http://192.168.2.66:32123/webdav/
vendor = other
user = <USER_NAME>
pass = <USER_PASSWD>
```

I tried a few more times and the fault of 1.61.0 does not exist on 1.60.1.

**Comment by wuhgit** [comment]
There was a problem with my previous description.
Accessing directly through the browser, I can see the list of files, but click any of these files shows `Not Found`

**Comment by darthShadow** [comment]
Any chance you could add the server & mount logs for both versions that have you listing the folder and then clicking on a file?

**Comment by ncw** [comment]
This is a bug in the `--baseurl` parameter. It works fine without it.

I'm investigating and I'll post an update shortly!

**Comment by ncw** [comment]
I think I've fixed this now - can you give this a try please

[v1.62.0-beta.6659.d6ffef088.fix-6650-serve-webdav-prefix](https://beta.rclone.org/branch/fix-6650-serve-webdav-prefix/v1.62.0-beta.6659.d6ffef088.fix-6650-serve-webdav-prefix/) on branch [fix-6650-serve-webdav-prefix](https://github.com/rclone/rclone/tree/fix-6650-serve-webdav-prefix) (uploaded in 15-30 mins)

**Comment by wuhgit** [comment]
> I think I've fixed this now - can you give this a try please
> 
> [v1.62.0-beta.6659.d6ffef088.fix-6650-serve-webdav-prefix](https://beta.rclone.org/branch/fix-6650-serve-webdav-prefix/v1.62.0-beta.6659.d6ffef088.fix-6650-serve-webdav-prefix/) on branch [fix-6650-serve-webdav-prefix](https://github.com/rclone/rclone/tree/fix-6650-serve-webdav-prefix) (uploaded in 15-30 mins)

@ncw 

Thank you so much! The problem is solved!

**Comment by ncw** [comment]
Thank you for testing. I'll merge this to master for v1.61.1 which I'll release today or tomorrow.

**Comment by ncw** [comment]
This fix has been released in v1.61.1 which is available now
